server/src/test/java/org/opensearch/common/cache/tier/TieredSpilloverCacheTests.java

```diff
@@ -0,0 +1,837 @@
+        assertEquals(1, onDiskCache.count());
+        assertNotNull(onDiskCache.get(keyToBeEvicted));
+    }
+
+    class MockCacheEventListener<K, V> implements StoreAwareCacheEventListener<K, V> {
+
+        EnumMap<CacheStoreType, TestStatsHolder> enumMap = new EnumMap<>(CacheStoreType.class);
+
+        MockCacheEventListener() {
+            for (CacheStoreType cacheStoreType : CacheStoreType.values()) {
+                enumMap.put(cacheStoreType, new TestStatsHolder());
+            }
+        }
+
+        @Override
+        public void onMiss(K key, CacheStoreType cacheStoreType) {
+            enumMap.get(cacheStoreType).missCount.inc();
+        }
+
+        @Override
+        public void onRemoval(StoreAwareCacheRemovalNotification<K, V> notification) {
+            if (notification.getRemovalReason().equals(RemovalReason.EVICTED)) {
+                enumMap.get(notification.getCacheStoreType()).evictionsMetric.inc();
+            } else if (notification.getRemovalReason().equals(RemovalReason.INVALIDATED)) {
+                enumMap.get(notification.getCacheStoreType()).invalidationMetric.inc();
+            }
+        }
+
+        @Override
+        public void onHit(K key, V value, CacheStoreType cacheStoreType) {
+            enumMap.get(cacheStoreType).hitCount.inc();
+        }
+
+        @Override
+        public void onCached(K key, V value, CacheStoreType cacheStoreType) {
+            enumMap.get(cacheStoreType).cachedCount.inc();
+        }
+
+        class TestStatsHolder {
+            final CounterMetric evictionsMetric = new CounterMetric();
+            final CounterMetric hitCount = new CounterMetric();
+            final CounterMetric missCount = new CounterMetric();
+            final CounterMetric cachedCount = new CounterMetric();
+            final CounterMetric invalidationMetric = new CounterMetric();
+        }
+    }
+
+    private LoadAwareCacheLoader<String, String> getLoadAwareCacheLoader() {
+        return new LoadAwareCacheLoader<String, String>() {
+            boolean isLoaded = false;
+
+            @Override
+            public String load(String key) {
+                isLoaded = true;
+                return UUID.randomUUID().toString();
+            }
+
+            @Override
+            public boolean isLoaded() {
+                return isLoaded;
+            }
+        };
+    }
+
+    private TieredSpilloverCache<String, String> intializeTieredSpilloverCache(
+        int onHeapCacheSize,
+        int diksCacheSize,
+        StoreAwareCacheEventListener<String, String> eventListener,
+        long diskDeliberateDelay
+    ) {
+        StoreAwareCacheBuilder<String, String> diskCacheBuilder = new MockOnDiskCache.Builder<String, String>().setMaxSize(diksCacheSize)
+            .setDeliberateDelay(diskDeliberateDelay);
+        StoreAwareCacheBuilder<String, String> onHeapCacheBuilder = new MockOnHeapCache.Builder<String, String>().setMaxSize(
```

Comment:
why are we using `MockOnHeapCache` here ? Isn't the purpose of this test to validate the `TieredSpilloverCache` with `OpenSearchOnHeapCache` implementation ?

 Also with mock based onDisk setup not sure what are we trying to verify with the tests. Given Tiered cache is expected to always have onHeap and disk based cache, probably we should test with those concrete implementations instead of mock.

Reply:
I have replaced MockOnHeapCache with actual OpenSearchOnHeapCache. 
Still using MockDisk as we don't have a actual implementation for disk yet. Will add later.